### ISSUE TITLE
[sql] JSON type in sql plugin

### ISSUE THREAD

**Issue by reedHam** [tauri-apps/plugins-workspace#93]
Hi, 

Thanks for your hard work.

I am trying to use the json datatype in an sqlite database. 

The values are inserted correctly but returned as null, due to the type being commented out of the select function.

https://github.com/tauri-apps/plugins-workspace/blob/d33ee156b865eddc162bcce28de7cc01db248efc/plugins/sql/src/plugin.rs#L283

This was the error receved when I uncommented the JSON data type. 
```
error[E0277]: the trait bound `serde_json::Map<std::string::String, JsonValue>: sqlx::Decode<'_, Sqlite>` is not satisfied
    |
283 |                     "JSON" => JsonValue::Object(row.get(i)),
    |                                                 ^^^ --- required by a bound introduced by this call
    |                                                 |
    |                                                 the trait `sqlx::Decode<'_, Sqlite>` is not implemented for `serde_json::Map<std::string::String, JsonValue>`
```

Any help is appreciated.

**Comment by FabianLars** [comment]
iirc we concluded that we don't want to implement direct json support to this plugin because the database itself doesn't support it. Instead we recommend using a TEXT/VARCHAR field and JSON.parse/JSON.stringify in the frontend (that seems to be the default solution for json in sqlite)

**Comment by reedHam** [comment]
Sqlite supports json as listed in the docs here. 
https://www.sqlite.org/json1.html
You cannot preform property extraction when storing JSON objects as strings so this limits functionality substantially.

**Comment by reedHam** [comment]
> iirc we concluded that we don't want to implement direct json support to this plugin because the database itself doesn't support it. Instead we recommend using a TEXT/VARCHAR field and JSON.parse/JSON.stringify in the frontend (that seems to be the default solution for json in sqlite)

Sqlite supports json as a datatype although it is stored as text, listed in the docs here.
https://www.sqlite.org/json1.html
And when I attempted to use it with the sql plugin I was able to store the information in the sqlite database but on retrieval a null value is returned. 
If the plugin is not going to support deserialization then rather than returning `NULL` for json datatypes returning the string representation or simply an error would be better I believe.
